PresentScreenings/Controllers/ViewController.cs

```diff
@@ -491,6 +511,116 @@ public void UpdateWarning(Screening screening)
+            var currDay = Plan.CurrDay;
+            var attendedScreenings = DayScreenings()
+                .Where(s => s.StartDate == currDay && s.StartTime < screening.EndTime && s.IAttend)
+                .OrderByDescending(s => s.EndTime)
+                .ToList();
+            attendedScreenings.Remove(screening);
+
+            // Find the first space where the given screening fits.
+            var otherScreenings = new List<Screening> { };
+            otherScreenings.AddRange(attendedScreenings);
+            foreach (var attendedScreening in attendedScreenings)
+            {
+                otherScreenings.Remove(attendedScreening);
+                if (otherScreenings.Count == 0)
+                {
+                    return attendedScreening.StartTime - screening.EndTime - _pause;
+                }
+                var nextScreening = otherScreenings[0];
+                if (attendedScreening.StartTime > nextScreening.EndTime + screening.Duration + 2 * _pause)
+                {
+                    return attendedScreening.StartTime - screening.EndTime - _pause;
+                }
+            }
+            return -_pause;
+        }
+
+        public void MoveTimeSpan(Screening screening, TimeSpan span)
+        {
+            if (screening is OnDemandScreening onDemandScreening)
+            {
+                onDemandScreening.MoveStartTime(span);
+                Plan.InitializeDays();
+                GoToDay(onDemandScreening.StartTime.Date);
+                //_mainView.HeadersView.DrawCurrDay(_plan);
+                //UpdateDayAttendanceStatus(screening);
+                //ReloadScreeningsView();
+                SetCurrScreening(screening);
+            }
+        }
+
+        public static bool MoveForwardAllowed(Screening screening, bool withinDay = false)
+        {
+            if (screening == null)
+            {
+                return false;
+            }
+            if (screening is OnDemandScreening onDemandScreening)
+            {
+                bool allowed = onDemandScreening.EndTime < onDemandScreening.WindowEndTime;
+                if (allowed && withinDay)
+                {
+                    allowed = screening.StartTime.TimeOfDay < new TimeSpan(23, 59, 0);
+                }
+                return allowed;
+            }
+            return false;
+        }
+
+        public static bool MoveBackwardAllowed(Screening screening, bool withinDay = false)
+        {
+            if (screening == null)
+            {
+                return false;
+            }
+            if (screening is OnDemandScreening onDemandScreening)
+            {
+                bool allowed = onDemandScreening.StartTime > onDemandScreening.WindowStartTime;
+                if (allowed && withinDay)
+                {
+                    allowed = screening.StartTime.TimeOfDay > new TimeSpan(9, 0, 0);
+                }
+                return allowed;
+            }
+            return false;
+        }
```

Comment:
I'm afraid such unification would not lead to less code, but would lead to less readable code instead.
This price is, in my opinion, not earned back in maintainability.
I did replace the magic numbers though.